examples/src/Components/FileSelect.tsx

```diff
@@ -16,57 +13,58 @@ interface FileSelectionProps {
 
 const FileSelection = ({
     selectedFile,
+    conversionFileName,
     onFileSelect,
     loadSmoldynFile,
     clearFile,
     loadSmoldynPreConfiguredSim,
     setRabbitCount,
 }: FileSelectionProps): JSX.Element => {
+    const [selectValue, setSelectValue] = React.useState("");
+
     useEffect(() => {
         const urlParams = new URLSearchParams(window.location.search);
         if (urlParams.has("file")) {
             const queryStringFile = urlParams.get("file") || "";
             onFileSelect(queryStringFile);
         }
-    }, [onFileSelect]);
+    }, []);
+
+    useEffect(() => {
+        if (conversionFileName !== "") {
+            setSelectValue("Awaiting file conversion...");
+        } else {
+            setSelectValue(selectedFile);
+        }
+    }, [selectedFile, conversionFileName]);
```

Comment:
This probably doesn't need to be a `useEffect` + state? 
```suggestion
const isAwaitingFileName = conversionFileName !== "";
const selectValue = isAwaitingFileName ? "Awaiting file conversion..." : selectedFile;
```